src/dmd/traits.d

```diff
@@ -627,6 +629,49 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         }
         return True();
     }
+    if (e.ident == Id.hasPostblit)
+    {
+        if (dim != 1)
+            return dimError(1);
+
+        auto o = (*e.args)[0];
+        auto t = isType(o);
+        if (!t)
+        {
+            e.error("type expected as second argument of __traits `%s` instead of `%s`",
+                e.ident.toChars(), o.toChars());
+            return new ErrorExp();
+        }
+
+        Type tb = t.baseElemOf();
+        if (auto sd = tb.ty == Tstruct ? (cast(TypeStruct)tb).sym : null)
+        {
+            return sd.postblit ? True() : False();
```

Comment:
and then here


```d
 return (e.ident == Id.hasPostblit) ? (sd.postblit ? True() : False())
                                    : (sd.hasCopyCtor ? True() : False());
```

Reply:
yeah i originally literally just mindlessly copy/pasted it but this is nicer, change made just waiting on the autotester to make sure i didnt mess up